public/app/features/dashboard-scene/pages/DashboardScenePageStateManager.ts

```diff
@@ -176,6 +189,32 @@ abstract class DashboardScenePageStateManagerBase<T>
     }
   }
 
+  public async loadProvisioningDashboard(repo: string, path: string): Promise<T> {
+    const params = new URLSearchParams(window.location.search);
+    const ref = params.get('ref') ?? undefined; // commit hash or branch
+
+    const url = `${BASE_URL}/repositories/${repo}/files/${path}`;
+    return getBackendSrv()
+      .get(url, ref ? { ref } : undefined)
+      .then((v) => {
```

Comment:
@Clarity-89 - this is where we need to introduce a proper typing for the response.